spec/lib/unique_within_region_validator_spec.rb

```diff
@@ -2,38 +2,33 @@
   describe "#unique_within_region" do
     context "class without STI" do
       let(:case_sensitive_class) do
-        Class.new(User).tap do |c|
-          c.class_eval do
-            validates :name, :unique_within_region => true
-          end
+        Class.new(ApplicationRecord) do
+          self.table_name = "users"
+          validates :name, :unique_within_region => true
         end
       end
 
       let(:case_insensitive_class) do
-        Class.new(User).tap do |c|
-          c.class_eval do
-            validates :name, :unique_within_region => {:match_case => false}
-          end
+        Class.new(ApplicationRecord) do
+          self.table_name = "users"
+          validates :name, :unique_within_region => {:match_case => false}
         end
       end
 
       let(:scoped_class) do
-        Class.new(User).tap do |c|
-          c.class_eval do
-            validates :name, :unique_within_region => {:scope => :email}
-          end
+        Class.new(ApplicationRecord) do
+          self.table_name = "users"
+          validates :name, :unique_within_region => {:scope => :email}
         end
       end
 
       let(:test_name)  { "thename" }
-      let(:test_email) { "thename@example.com" }
 
       let(:in_first_region_id) do
         FactoryBot.create(
           :user,
```

Comment:
If we're no longer inheriting from the `User` class above, should these stop using the User factory?

Reply:
I was unimpressed that it used the user factory

Also did not want to test anything that was outside this file. It tends to make stale tests quickly

I didn't know the proper way for creating these. I guess I could just use `Model.create`?

Reply:
Yeah, `Model.create` sounds better.